File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    arr_len = len(arr)
    el_len = len(arr[0])
    
    if arr_len > el_len:
        for ar in arr:
            ar += [0 for _ in range(arr_len - el_len)]
    elif arr_len < el_len:
        for _ in range(el_len - arr_len):
            arr.append([0 for _ in range(el_len)])

    return arr

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.java
import java.util.Arrays;

class Solution {
    public int[][] solution(int[][] arr) {
        int arrLen = arr.length;
        int elLen = arr[0].length;
        
        if (arrLen > elLen) {
            for (int i = 0; i < arrLen; i++) {
                arr[i] = Arrays.copyOf(arr[i], arrLen);
            }
        } else if (arrLen < elLen) {
            int[][] newArr = new int[elLen][elLen];
            for (int i = 0; i < arrLen; i++) {
                newArr[i] = Arrays.copyOf(arr[i], elLen);
            }
            for (int i = arrLen; i < elLen; i++) {
                Arrays.fill(newArr[i], 0);
            }
            arr = newArr;
        }
        
        return arr;
    }
}